pandas/core/indexing.py

```diff
@@ -2077,6 +2072,8 @@ def _getitem_tuple(self, tup: Tuple):
 
             # if the dim was reduced, then pass a lower-dim the next time
             if retval.ndim < self.ndim:
+                # TODO: this is never reached in tests; can we confirm that
+                #  it is impossible?
```

Comment:
@toobaz any thoughts on showing this is unreachable?  If we can confirm this is the case, then this chunk of the code can be shared with the other _getitem_tuple method above

(pinging you since you wrote the wiki page on simplifying this code)

Reply:
looks like this also wasnt hit in 0.24.2

Reply:
This path was *exercised* by the test suite in the commit in which it was introduced, but not by the tests added in that commit. To me it looks like most of these test cases still exist, but no longer hit the path.

test log and reproduction info: https://pastebin.com/qxjdpyd3

Reply:
And for context, example case which triggers on that commit but not on master

```python
import numpy as np
from pandas import *


mi_int = DataFrame(
    np.random.randn(3, 3),
    columns=[[2, 2, 4], [6, 8, 10]],
    index=[[4, 4, 8], [8, 10, 12]],
)

# this triggers the code path when testing against 9f0dc3befb
rs = mi_int.iloc[:, 2]  
```

Reply:
I don't think that path is ever reached. It should be reached by something like ``df.loc[1, [0, 1]]``, which however gets captured in the call above to ``self._getitem_lowerdim(tup)`` (which doesn't make sense, because it is named "lowerdim", but then it resorts to ``_is_nested_tuple_indexer``). So to sum up: I don't know exactly what the aim of that ``axis -= 1`` is, I don't think it is ever reached, but I think it _should_ be reached in principle.

Reply:
(but no major objections to removing for now)

Reply:
@pilkibun thanks for tracking that down

@toobaz how about we fix getitem_lowerdim to make sense before removing this check.  simplifying this code is going to take a few passes

Reply:
@pilkibun there is another un-hit block on lines 440-450.  any idea what that was intended to catch?

Reply:
this was related to Panel indexing, can probably go.

Reply:
@jbrockmendel, the code was introduced in 60efe85bc2ce9578e8f5e0c8 https://github.com/pandas-dev/pandas/blob/60efe85bc2ce9578e8f5e0c88907c5cc6b6b7b31/pandas/core/indexing.py#L264-L275

At that time there was a single test that entered the top conditional:
https://github.com/pandas-dev/pandas/blob/60efe85bc2ce9578e8f5e0c88907c5cc6b6b7b31/pandas/tests/test_indexing.py#L971

but it immediately triggered the ValueError condition. No test reached the code below. 

The shrunk original test case was:

<details><summary>Code</summary>

```python

from pandas import *

df = DataFrame(
    np.random.randint(5, 10, size=9).reshape(3, 3),
    columns=list("abc"),
    index=[[4, 4, 8], [8, 10, 12]],
)

df.ix[4, "c"] = [0, 1, 2, 3]
```

</details>


The test still exists in master but has been modified to use loc: https://github.com/pandas-dev/pandas/blob/5b1a870a057a7f05d2967aae119e144cb17a0665/pandas/tests/indexing/multiindex/test_setitem.py#L205-L208
and it still triggers the same `ValueError` condition.

Reply:
> @toobaz how about we fix getitem_lowerdim to make sense before removing this check. simplifying this code is going to take a few passes

No objection